x-pack/elastic-agent/pkg/agent/application/monitoring_decorator.go

```diff
@@ -65,8 +65,21 @@ func injectMonitoring(agentInfo *info.AgentInfo, outputGroup string, rootAst *tr
 		monitoringOutputName = monitoringOutputNameKey.String()
 	}
 
+	typeValue := elasticsearchKey
+	t, found := transpiler.Lookup(rootAst, fmt.Sprintf("%s.%s.type", outputsKey, monitoringOutputName))
+	if found {
+		v, ok := t.Value().(*transpiler.StrVal)
+		if !ok {
+			return programsToRun, nil
```

Comment:
Shouldn't this be an error? It looks we hit this line if someone did not follow the config format schema. If so we should rather fail instead of continuing with some 'internal' state that is difficult to understand for the user.

Reply:
This indeed should error, I've tried refactoring in this PR but it became a mess, I will create a followup issue so we can consider this as techdebt and refactor it.